modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationEndpoints/CreateAnAPIApplicationEndpoint.testcase

```diff
@@ -94,19 +114,274 @@ definition {
+
+			AssertElementPresent(
+				key_fieldLabel = "Path",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "path-defaul");
+
+			AssertElementPresent(
+				key_value = "endpoint",
+				locator1 = "APIBuilder#DESCRIPTION",
+				value1 = "Description default");
+		}
+	}
+
+	@priority = 4
+	test CanSeePathOfPathFields {
+		property portal.acceptance = "true";
+
+		task ("When I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("And Given I click 'Add API Endpoint' button") {
+			Click(locator1 = "Button#PLUS");
+		}
+
+		task ("Then I can see path under {portalUrl}/o/c/{applicationName}") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "APIBuilder#URL_LABEL",
+				value1 = "http://liferay-f0g7f0h7:8080/o/c/test2");
+		}
+
+		task ("And Then I can see path start with '/' with placeholder 'Enter path'") {
+			AssertElementPresent(locator1 = "APIBuilder#SLASH");
+		}
+	}
+
+	@priority = 4
+	test CanSeeScopeOfScopeFields {
+		property portal.acceptance = "true";
+
+		task ("And Given I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("And Given I click 'Add API Endpoint' button") {
+			Click(locator1 = "Button#PLUS");
+		}
+
+		task ("When I click 'Create' button in 'New API Endpoint'") {
+			Click(
+				key_value = "Select Scope",
+				locator1 = "APIBuilder#CREATE_BUTTON");
+		}
+
+		task ("Then Then I click Scope option") {
+			Click(
+				key_value = "Select Scope",
+				locator1 = "APIBuilder#DROPDOWN_TOGGLE");
+		}
+
+		task ("And Then I can see site and company scope in the Scope option") {
+			AssertElementPresent(
+				key_optionName = "Site",
+				locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+
+			AssertElementPresent(
+				key_optionName = "Company",
+				locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+		}
+	}
+
+	@priority = 4
+	test CanSeeValidationMessageForRequiredFields {
+		property portal.acceptance = "true";
+
+		task ("And Given I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("And Given I click 'Add API Endpoint' button") {
+			Click(locator1 = "Button#PLUS");
+		}
+
+		task ("When I click 'Create' button in 'New API Endpoint'") {
+			Click(locator1 = "APIBuilder#CREATE_BUTTON");
+		}
+
+		task ("Then I can see error message about required Scope and Path") {
+			AssertElementPresent(
+				locator1 = "CommerceEntry#PRODUCT_DETAIL_UUID",
+				uuid = "Please select a scope");
+
+			AssertElementPresent(
+				locator1 = "CommerceEntry#PRODUCT_DETAIL_UUID",
+				uuid = "Please select a scope");
+		}
+
+		task ("And Then no endpoint is being created") {
+			Click(locator1 = "CP2Utils#CLOSE_BUTTON");
+
```

Comment:
you can try to use 'Button#CANCEL' to replace click close button, if you want to click close button, you can try to use 'Modal#CLOSE_BUTTON', thanks!

Reply:
Fixed, Thanks!